File: node/session_test.js
const jlibtorrent = require('./jlibtorrent.js');

console.log("Using libtorrent version: " + jlibtorrent.LibTorrent.fullVersion());

const sp = new jlibtorrent.SettingsPack();
const s = new jlibtorrent.Session(sp, false);

s.on('alert', function (a) {
    console.log(a.type() + " - " + a.what() + " - " + a.message());
});

process.stdout.write('Press ENTER to exit...');
process.stdin.once('data', function (data) {
    process.exit(0);
});

File: src/test/java/com/frostwire/jlibtorrent/demo/SessionTest.java
package com.frostwire.jlibtorrent.demo;

import com.frostwire.jlibtorrent.AlertListener;
import com.frostwire.jlibtorrent.LibTorrent;
import com.frostwire.jlibtorrent.Session;
import com.frostwire.jlibtorrent.SettingsPack;
import com.frostwire.jlibtorrent.alerts.Alert;

import java.util.Map;

/**
 * @author gubatron
 * @author aldenml
 */
public final class SessionTest {

    public static void main(String[] args) throws Throwable {

        AlertListener l = new AlertListener() {
            @Override
            public int[] types() {
                return null;
            }

            @Override
            public void alert(Alert<?> alert) {
                System.out.println(alert);
            }
        };

        Session s = new Session(new SettingsPack(), false, l);

        System.out.println("Press ENTER to exit");
        System.in.read();

        s.abort();
    }
}
